<a href="https://colab.research.google.com/github/Guioliveiraa/qui/blob/main/(GuilhermeOliveira)_semana11.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
import pandas as pd


In [4]:
arq_csv = "https://dados.ufs.br/dataset/cursos/resource/cur-csv-cursos-da-ufs/download/cur-csv-cursos-da-ufs.csv"

In [5]:
df_arq = pd.read_csv (arq_csv)

In [6]:
df_arq

nome  ... alunos_concluidos
0    ADMINISTRAÇÃO  ...               698
1    ADMINISTRAÇÃO  ...                 1
2    ADMINISTRAÇÃO  ...               791
3    ADMINISTRAÇÃO  ...                 1
4    ADMINISTRAÇÃO  ...               156
..             ...  ...               ...
397        TURISMO  ...               161
398      ZOOTECNIA  ...                 0
399      ZOOTECNIA  ...                88
400      ZOOTECNIA  ...                37
401      ZOOTECNIA  ...               159

[402 rows x 21 columns]

In [ ]:
df_arq ["alunos_ativos"].max()

554

CURSOS COM ALUNOS ATIVOS MAIOR QUE 100

In [ ]:
df_arq [df_arq ["alunos_ativos"] > 100] 

nome  ... alunos_concluidos
0          ADMINISTRAÇÃO  ...               698
3          ADMINISTRAÇÃO  ...                 1
4          ADMINISTRAÇÃO  ...               156
11         AGROINDUSTRIA  ...                26
16           ARQUEOLOGIA  ...               169
..                   ...  ...               ...
395               TEATRO  ...                61
396  TERAPIA OCUPACIONAL  ...               148
397              TURISMO  ...               161
400            ZOOTECNIA  ...                37
401            ZOOTECNIA  ...               159

[111 rows x 21 columns]

CURSOS NOTURNOS


In [ ]:
df_arq [df_arq ["turno"] == "Noturno"]

nome  ... alunos_concluidos
0                     ADMINISTRAÇÃO  ...               698
4                     ADMINISTRAÇÃO  ...               156
21   BIBLIOTECONOMIA E DOCUMENTAÇÃO  ...               204
41               CIÊNCIAS ATUARIAIS  ...                57
53              CIÊNCIAS BIOLÓGICAS  ...               277
57               CIÊNCIAS CONTÁBEIS  ...               283
58               CIÊNCIAS CONTÁBEIS  ...              1432
60             CIÊNCIAS DA RELIGIÃO  ...                51
64              CIÊNCIAS ECONÔMICAS  ...               475
87                            DANÇA  ...               117
91                   DESIGN GRÁFICO  ...                91
94                          DIREITO  ...              1149
151                     ESTATÍSTICA  ...               140
159                       FILOSOFIA  ...               333
166                          FÍSICA  ...               294
168                          FÍSICA  ...                96
203                       GEOGRAFIA  ...               132
219                        HISTÓRIA  ...               407
241                          LETRAS  ...               290
243                          LETRAS  ...                92
244                          LETRAS  ...                 9
249                          LETRAS  ...               535
254                          LETRAS  ...               343
282      LETRAS - LÍNGUA PORTUGUESA  ...               327
311                      MATEMÁTICA  ...               232
324                          MÚSICA  ...               160
340                       PEDAGOGIA  ...               461
341                       PEDAGOGIA  ...               343
357                         QUÍMICA  ...               298
383          SECRETARIADO EXECUTIVO  ...               201
386                  SERVIÇO SOCIAL  ...               611
387                  SERVIÇO SOCIAL  ...                 2
390          SISTEMAS DE INFORMAÇÃO  ...               139
394                          TEATRO  ...                37
395                          TEATRO  ...                61

[35 rows x 21 columns]

CURSOS NOTURNOS COM NUMERO DE ALUNOS MAIOR QUE 100


In [ ]:
df_arq [df_arq ["alunos_ativos"] > 100 ] [df_arq ["turno"] == "Noturno"]

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  """Entry point for launching an IPython kernel.


nome  ... alunos_concluidos
0                     ADMINISTRAÇÃO  ...               698
4                     ADMINISTRAÇÃO  ...               156
21   BIBLIOTECONOMIA E DOCUMENTAÇÃO  ...               204
41               CIÊNCIAS ATUARIAIS  ...                57
53              CIÊNCIAS BIOLÓGICAS  ...               277
57               CIÊNCIAS CONTÁBEIS  ...               283
58               CIÊNCIAS CONTÁBEIS  ...              1432
60             CIÊNCIAS DA RELIGIÃO  ...                51
64              CIÊNCIAS ECONÔMICAS  ...               475
91                   DESIGN GRÁFICO  ...                91
94                          DIREITO  ...              1149
151                     ESTATÍSTICA  ...               140
159                       FILOSOFIA  ...               333
166                          FÍSICA  ...               294
168                          FÍSICA  ...                96
203                       GEOGRAFIA  ...               132
219                        HISTÓRIA  ...               407
241                          LETRAS  ...               290
243                          LETRAS  ...                92
249                          LETRAS  ...               535
254                          LETRAS  ...               343
282      LETRAS - LÍNGUA PORTUGUESA  ...               327
311                      MATEMÁTICA  ...               232
324                          MÚSICA  ...               160
340                       PEDAGOGIA  ...               461
341                       PEDAGOGIA  ...               343
357                         QUÍMICA  ...               298
383          SECRETARIADO EXECUTIVO  ...               201
386                  SERVIÇO SOCIAL  ...               611
390          SISTEMAS DE INFORMAÇÃO  ...               139
395                          TEATRO  ...                61

[31 rows x 21 columns]

CURSOS VESPERTINOS COM NUMERO DE ALUNOS MAIOR QUE 100


In [ ]:
df_arq [df_arq ["turno"] == "Vespertino"][df_arq ["alunos_ativos"] > 100 ] 

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  """Entry point for launching an IPython kernel.


nome  ... alunos_concluidos
3                          ADMINISTRAÇÃO  ...                 1
16                           ARQUEOLOGIA  ...               169
19                         ARTES VISUAIS  ...               267
33                 CIÊNCIA DA COMPUTAÇÃO  ...               525
54                   CIÊNCIAS BIOLÓGICAS  ...               359
55                   CIÊNCIAS BIOLÓGICAS  ...               502
65                   CIÊNCIAS ECONÔMICAS  ...               546
71                      CIÊNCIAS SOCIAIS  ...               207
73                  CINEMA E AUDIOVISUAL  ...                43
96                               DIREITO  ...              1258
109                      EDUCAÇÃO FÍSICA  ...               245
125                     ENGENHARIA CIVIL  ...              1161
127             ENGENHARIA DE COMPUTAÇÃO  ...                 2
129              ENGENHARIA DE MATERIAIS  ...                94
130                  ENGENHARIA DE PESCA  ...               120
132               ENGENHARIA DE PRODUÇÃO  ...               226
152                             FARMÁCIA  ...               237
173                               FÍSICA  ...                81
178                  FÍSICA: ASTROFÍSICA  ...                 6
179                FÍSICA: FÍSICA MÉDICA  ...                 9
204                            GEOGRAFIA  ...               366
246                               LETRAS  ...               293
297                           MATEMÁTICA  ...               179
309                           MATEMÁTICA  ...               453
315  MATEMÁTICA APLICADA E COMPUTACIONAL  ...                10
336                            PEDAGOGIA  ...              1061
349                           PSICOLOGIA  ...               650
354             PUBLICIDADE E PROPAGANDA  ...                 3
356                              QUÍMICA  ...               133
375              RELAÇÕES INTERNACIONAIS  ...               141
397                              TURISMO  ...               161

[31 rows x 21 columns]

In [ ]:
df_arq [df_arq ["turno"] == "Vespertino"]

nome  ... alunos_concluidos
3                          ADMINISTRAÇÃO  ...                 1
16                           ARQUEOLOGIA  ...               169
19                         ARTES VISUAIS  ...               267
33                 CIÊNCIA DA COMPUTAÇÃO  ...               525
54                   CIÊNCIAS BIOLÓGICAS  ...               359
55                   CIÊNCIAS BIOLÓGICAS  ...               502
65                   CIÊNCIAS ECONÔMICAS  ...               546
71                      CIÊNCIAS SOCIAIS  ...               207
72                      CIÊNCIAS SOCIAIS  ...               336
73                  CINEMA E AUDIOVISUAL  ...                43
80                    COMUNICAÇÃO SOCIAL  ...               208
81                    COMUNICAÇÃO SOCIAL  ...                89
83                    COMUNICAÇÃO SOCIAL  ...               161
96                               DIREITO  ...              1258
109                      EDUCAÇÃO FÍSICA  ...               245
112                      EDUCAÇÃO FÍSICA  ...               969
117                           ENFERMAGEM  ...               183
125                     ENGENHARIA CIVIL  ...              1161
127             ENGENHARIA DE COMPUTAÇÃO  ...                 2
129              ENGENHARIA DE MATERIAIS  ...                94
130                  ENGENHARIA DE PESCA  ...               120
132               ENGENHARIA DE PRODUÇÃO  ...               226
152                             FARMÁCIA  ...               237
154                             FARMÁCIA  ...               299
155                             FARMÁCIA  ...                24
156                             FARMÁCIA  ...                20
167                               FÍSICA  ...                 2
173                               FÍSICA  ...                81
178                  FÍSICA: ASTROFÍSICA  ...                 6
179                FÍSICA: FÍSICA MÉDICA  ...                 9
204                            GEOGRAFIA  ...               366
218                             HISTÓRIA  ...                13
246                               LETRAS  ...               293
297                           MATEMÁTICA  ...               179
309                           MATEMÁTICA  ...               453
310                           MATEMÁTICA  ...                41
315  MATEMÁTICA APLICADA E COMPUTACIONAL  ...                10
336                            PEDAGOGIA  ...              1061
349                           PSICOLOGIA  ...               650
354             PUBLICIDADE E PROPAGANDA  ...                 3
356                              QUÍMICA  ...               133
358                              QUÍMICA  ...               217
371                  QUÍMICA TECNOLÓGICA  ...                 5
375              RELAÇÕES INTERNACIONAIS  ...               141
397                              TURISMO  ...               161

[45 rows x 21 columns]

CURSOS BACHARELADO


In [7]:
df_arq [df_arq ["grau_academico"] == "Bacharelado"]

nome  ... alunos_concluidos
0             ADMINISTRAÇÃO  ...               698
2             ADMINISTRAÇÃO  ...               791
3             ADMINISTRAÇÃO  ...                 1
4             ADMINISTRAÇÃO  ...               156
6     ADMINISTRAÇÃO PÚBLICA  ...               115
..                      ...  ...               ...
391  SISTEMAS DE INFORMAÇÃO  ...                74
396     TERAPIA OCUPACIONAL  ...               148
397                 TURISMO  ...               161
400               ZOOTECNIA  ...                37
401               ZOOTECNIA  ...               159

[101 rows x 21 columns]

CURSOS DE LICENCIATURA PLENA



In [8]:
df_arq [df_arq ["grau_academico"] == "Licenciatura Plena"]


nome  ... alunos_concluidos
1          ADMINISTRAÇÃO  ...                 1
19         ARTES VISUAIS  ...               267
42   CIÊNCIAS BIOLÓGICAS  ...                 9
43   CIÊNCIAS BIOLÓGICAS  ...                 2
44   CIÊNCIAS BIOLÓGICAS  ...                 3
..                   ...  ...               ...
366              QUÍMICA  ...                 0
367              QUÍMICA  ...                 1
368              QUÍMICA  ...                 1
394               TEATRO  ...                37
395               TEATRO  ...                61

[165 rows x 21 columns]

GUILHERME OLIVEIRA TORRES GOMES 0050016114




Existem mais cursos de Licenciatura do que de bacharelado

O numero maximo de alunos ativos em um curso é de 554

